code/src/pages/MovieDetail.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from 'react'
+import { useParams, Link } from 'react-router-dom'
+
+import arrow from "../components/back-arrow.png"
+
+export const MovieDetail = () => {
+  const { id } = useParams()
+  const [movieDetail, setMovieDetail] = useState ([])
+  const MOVIE_DETAIL_URL = `https://api.themoviedb.org/3/movie/${id}?api_key=2c2891737dd5ffa64a8858cdba25ddfe&language=en-US`
+
+  useEffect(() => {
+    fetch(MOVIE_DETAIL_URL)
+    .then((res) => res.json())
+    .then((json) => {
+      setMovieDetail(json)
+      console.log(json) 
+    })
+  }, [MOVIE_DETAIL_URL])
+
+  return (
+    <div> {movieDetail.length===0 ?<></> :    
```

Comment:
We are not sure what this do, could you explain to us? *two lost sheeps* 🐑 🐑

Reply:
This prevents the page to go white (not found) when the user refreshes the page. ;)